pkg/virt-handler/vm.go

```diff
@@ -955,14 +955,18 @@ func (d *VirtualMachineController) updateInterfacesFromDomain(vmi *v1.VirtualMac
 			// Remove the interface from domainInterfaceStatusByMac to mark it as handled
 			if interfaceStatus, exists := domainInterfaceStatusByMac[interfaceMAC]; exists {
 				newInterface.InterfaceName = interfaceStatus.InterfaceName
+				newInterface.InfoSource = interfaceStatus.InfoSource
```

Comment:
> If I understand correctly, this info may include the fact that the source is guest-agent only or both guest-agent and domain. Is this correct?

It will look on `domainInterfaceStatusByMac` (derived from `domain.Status.Interfaces`) and take InfoSource from the,
`InfoSource` was calculated on the merge function of virt-launcher (and can be either one of the 3 options)

Reply:
> There are actually 3 places, because virt-launcher merge is called from two places in virt-launcher.
(the one where launcher push data, and the one that is called every 300 seconds via GetDomain)

Thank you for the clarification.
Another question: Is this data populated if there is no guest-agent?

Reply:
> Another question: Is this data populated if there is no guest-agent?

Yes, the domain.spec will be populated, once in 300 seconds via GetDomain (virt-handler initiated the request every 300s), or upon every trigger of virt-launcher domain modify event (if there are any when there is no guest-agent data)